pkg/gitops/fileprocessor/gotemplates.go

```diff
@@ -0,0 +1,55 @@
+package fileprocessor
+
+import (
+	"bytes"
+	"github.com/pkg/errors"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+const (
+	templateExtension = ".templ"
+)
+
+// TemplateParameters represents the API variables that can be used to template a profile. This set of variables will
+// be applied to the go template files found. Templates filenames must end in .templ
+type TemplateParameters struct {
+	ClusterName string
+}
+
+// NewTemplateParameters creates a set of variables for templating given a ClusterConfig object
+func NewTemplateParameters(clusterConfig *api.ClusterConfig) TemplateParameters {
+	return TemplateParameters{
+		ClusterName: clusterConfig.Metadata.Name,
+	}
+}
+
+// GoTemplateProcessor is a FileProcessor that executes Go Templates
+type GoTemplateProcessor struct {
+	Params TemplateParameters
```

Comment:
Is there a reason why we "pass" the `File` as an argument, but the `TemplateParameters` as a field?
If feels a bit strange to me, given they the two operands of a "generate manifest" operation. So I'd rather have them both passe to `ProcessFile`.

Reply:
The reason is that the templates are stable for all the processing since they are our API so you only need one value of it. Then `processFile()` is called once per file with the same contents of the TemplateParameters. Does that make more sense?

Reply:
Fair enough 👍